core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1132,6 +1157,38 @@ public VersionNumber getNeededDependenciesRequiredCore() {
             return versionNumber;
         }
 
+        /**
+         * Returns the minimum Java version needed to use the plugin and all its dependencies.
+         *
+         * @return the minimum Java version needed to use the plugin and all its dependencies.
+         *
+         * @since TODO
+         */
+        @CheckForNull
+        public VersionNumber getNeededDependenciesMinimumJavaVersion() {
+            VersionNumber versionNumber = null;
+            try {
+                versionNumber = minimumJavaVersion == null ? null : new VersionNumber(minimumJavaVersion);
+            } catch (NumberFormatException nfe) {
+                logBadMinJavaVersion(minimumJavaVersion);
+            }
+            for (Plugin p: getNeededDependencies()) {
+                VersionNumber v = p.getNeededDependenciesMinimumJavaVersion();
+                if (v == null) {
+                    continue;
+                }
+                if (versionNumber == null || v.isNewerThan(versionNumber)) {
+                    versionNumber = v;
+                }
+            }
+            return versionNumber;
+        }
+
+        private void logBadMinJavaVersion(String minimumJavaVersion) {
+            LOGGER.log(Level.WARNING, "'minimumJavaVersion' was specified for plugin ''{0}'' but unparseable (received {1})",
```

Comment:
Might have missed a doubling of `'` here. I recommend just avoiding ASCII apostrophes in message formats. Use `‘…’` or nothing.